src/App.tsx

```diff
@@ -15,36 +16,98 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
-export const App: React.FC = () => {
+enum SortType {
+  default = '',
+  byAlphabet = 'alphabet',
+  byLength = 'length',
+}
+
+function getPreparedGoods(
+  goods: string[],
+  sortField: SortType,
+  reversed: boolean,
+) {
+  const preparedGoods = [...goods];
+
+  if (sortField) {
+    preparedGoods.sort((good1, good2) => {
+      switch (sortField) {
+        case SortType.byAlphabet:
+          return good1.localeCompare(good2);
+        case SortType.byLength:
+          return good1.length - good2.length;
+        default:
+          return 2;
+      }
+    });
+  }
+
+  if (reversed) {
+    preparedGoods.reverse();
+  }
+
+  return preparedGoods;
+}
+
+export const App = () => {
+  const [sortField, setSortField] = useState(SortType.default);
+  const [reversed, setReversed] = useState(false);
+  const visibleGoods = getPreparedGoods(goodsFromServer, sortField, reversed);
+
+  function handleReset() {
```

Comment:
According to the [CODE STYLE] checklist, function names should start with a verb. Consider renaming `handleReset` to `resetSortAndOrder` or similar to clearly describe its action.